### PR TITLE
bulkio: Add scheduled jobs monitoring metrics.

### PR DESCRIPTION
**Pull request by miretskiy**
Informs #51847 

Add scheduled jobs monitoring metrics.

The scheduler daemon exports metrics related to each
execution loop, as well as the total counts on the number
of schedules executed, succeeded or failed.
Each executor may also optionally export executor specific metrics.

In addition, improve logging when executing schedules.

Release Notes: None

### CODE REVIEW

**Comment by miretskiy** [comment]
Friendly ping, @dt

File: pkg/jobs/schedule_metrics.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+	RescheduleSkip *metric.Gauge
+	// Number of schedules rescheduled due to WAIT policy.
+	RescheduleWait *metric.Gauge
+	// Number of schedules that could not be processed due to an error.
+	NumBadSchedules *metric.Counter
+}
+
+// MakeSchedulerMetrics returns metrics for scheduled job daemon.
+func MakeSchedulerMetrics() SchedulerMetrics {
+	return SchedulerMetrics{
```
Comment (dt) [inline pkg/jobs/schedule_metrics.go:50]: I'm a little skeptical we'll actually look at / act on the per-round gauges, and timeseries are not free, but as long as it is just a few i guess it isn't too bad.

File: pkg/jobs/job_scheduler.go [hunk 2]
```diff
@@ -135,23 +164,76 @@ func (s *jobScheduler) processSchedule(
...
+	if err != nil {
+		return nil, err
+	}
+	if m := ex.Metrics(); m != nil {
+		s.registry.AddMetricStruct(m)
+	}
+	return ex, nil
+}
+
+func newLoopStats(
```
Comment (dt) [inline pkg/jobs/job_scheduler.go:201]: I'm somewhat skeptical these are actually worth collecting / maintaining by default, but we can always remove them after 20.2 if we found we never used them, so I'm no opposed to adding them for now. I just might leave a TODO to come back and see if we ever actually acted on these or if we could avoid these extra scan / timeseries overhead/downsamples/etc.

**Review by dt — APPROVED** [review]
nit: "meetrics" in commit/PR title.

**Review by miretskiy — COMMENTED** [review]
Ty. Fixed

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/52741#-:-MEy-B-h2PSuc53h3Ful:bjvem9k)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dt)

___
*[pkg/jobs/job_scheduler.go, line 201 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52741#-MEZnQ7xmg8kPgf4Gmce-r1-201:-MEy-Dxh-mH-M4_WrPNw:bnuyih0) ([raw file](https://github.com/cockroachdb/cockroach/blob/27c049aac9753d99e934e115024c42ddcb202087/pkg/jobs/job_scheduler.go#L201)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary><blockquote>

I'm somewhat skeptical these are actually worth collecting / maintaining by default, but we can always remove them after 20.2 if we found we never used them, so I'm no opposed to adding them for now. I just might leave a TODO to come back and see if we ever actually acted on these or if we could avoid these extra scan / timeseries overhead/downsamples/etc.
</blockquote></details>

I personally found them useful.  They tell you if the scheduler runs correctly.
Also, given that it's a new system, i'd rather err on the side of adding more metrics.
But, as you said, we can remove later.
TODO added.
___
*[pkg/jobs/schedule_metrics.go, line 50 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/52741#-MEZnQ7xmg8kPgf4Gmcg-r1-50:-MEy-_DtAoyzu73IxqHA:b-5kygrp) ([raw file](https://github.com/cockroachdb/cockroach/blob/27c049aac9753d99e934e115024c42ddcb202087/pkg/jobs/schedule_metrics.go#L50)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary><blockquote>

I'm a little skeptical we'll actually look at / act on the per-round gauges, and timeseries are not free, but as long as it is just a few i guess it isn't too bad.
</blockquote></details>

Added todo, as you suggested.

**Pull request merged**